doc/hooks/versionCheckHook.section.md

```diff
@@ -33,6 +33,8 @@ The variables that this phase control are:
 - `dontVersionCheck`: Disable adding this hook to the [`preInstallCheckHooks`](#ssec-installCheck-phase). Useful if you do want to load the bash functions of the hook, but run them differently.
 - `versionCheckProgram`: The full path to the program that should print the `${version}` string. Defaults roughly to `${placeholder "out"}/bin/${pname}`. Using `$out` in the value of this variable won't work, as environment variables from this variable are not expanded by the hook. Hence using `placeholder` is unavoidable.
 - `versionCheckProgramArg`: The argument that needs to be passed to `versionCheckProgram`. If undefined the hook tries first `--help` and then `--version`. Examples: `version`, `-V`, `-v`.
+- `versionCheckKeepEnvironment`: A list of environment variables to keep and pass to the command.
+- `versionCheckDontIgnoreEnvironment`: Don't use `--ignore-environment` and pass all environment variables to the command. Prefer `versionCheckKeepEnvironment` if possible.
```

Comment:
I would reword this differently, perhaps "using knowingly" ?

Reply:
Sorry, I'm not sure I get what you mean. How can you use one of these parameters unknowingly?

Reply:
What I mean is that it is perhaps better to describe what those attributes are doing without mentioning to not use one or another.
If at some point we need to deprecate one of them, we will have to add a warning or a throw at the right place.

Reply:
> What I mean is that it is perhaps better to describe what those attributes are doing without mentioning to not use one or another.

They have very closely related meaning, so I don't find that mentioning too bad. However you do have a point, and that makes me suggest merging them into a single setting - maybe such that a `versionCheckKeepEnvironment = "all";` would be treated like `versionCheckDontIgnoreEnvironment = true`? What do you think @defelo ? I should also note that it shouldn't be too bad to perform a seemingly breaking change for `versionCheckDontIgnoreEnvironment`, as this attribute support is very new.

Reply:
Yes, I think merging these two parameters would be a good idea. But maybe we could use something like `versionCheckKeepEnvironment = "*";` (which should not be a valid environment variable name) instead of `"all"`.

However I still think we should at least mention that only those variables should be kept that are actually needed instead of just keeping all of them. In most cases it's probably just `[ "HOME" ]` anyways.

Reply:
@drupol @doronbehar I merged them now and adjusted the only use of `versionCheckDontIgnoreEnvironment` so far.